scripts/leapfrog/generate_resources.py

```diff
@@ -0,0 +1,322 @@
+        password=conf["password"],
+        project_name=conf["project_name"],
+        user_domain_id=conf["user_domain_id"],
+        project_domain_id=conf["project_domain_id"],
+    )
+
+    return session_.Session(auth=auth, verify=True)
+
+
+def session_caching_wrapper():
+    cached = {}
+
+    def get_session():
+        if not cached.get("session"):
+            conf = get_conf()
+            cached["session"] = _get_session(conf["auth"])
+        return cached["session"]
+    return get_session
+
+
+get_session = session_caching_wrapper()
+
+
+def get_flavor_id_from_name(name):
+    nova = nova_client.Client(CLIENT_VERSION, session=get_session())
+    flavors = [
+        flavor for flavor in nova.flavors.list() if flavor.name == name
+    ]
+    if len(flavors) > 1:
+        raise Exception("Flavor name, '{}', is not unique.".format(name))
+    try:
+        return flavors[0].id
+    except IndexError:
+        raise Exception("No flavor with name '{}' exists.".format(name))
+
+
+def get_network_id_from_name(name):
+    neutron = neutron_client.Client(session=get_session())
+    networks_ = neutron.list_networks(name=name)
+    networks = networks_["networks"]
+    if len(networks) > 1:
+        raise Exception("Network name, '{}', is not unique.".format(name))
+    try:
+        return networks[0]["id"]
+    except IndexError:
+        raise Exception("No network with name '{}' exists.".format(name))
+
+
+def get_image_id_from_name(name):
+    glance = glance_client.Client(CLIENT_VERSION, session=get_session())
+    images = list(glance.images.list(name=name))
+    if len(images) > 1:
+        raise Exception("Image name, '{}', is not unique.".format(name))
+    try:
+        return images[0].id
+    except IndexError:
+        raise Exception("No image with name '{}' exists.".format(name))
+
+
+def verify_active(building, timeout=600):
+    nova = nova_client.Client(CLIENT_VERSION, session=get_session())
+    active = []
+    counter = 0
+    delay = 10
+    while building:
+        still_building = []
+        for server in building:
+            server = nova.servers.get(server.id)
+            if server.status == "ACTIVE":
+                active.append(server)
+            elif server.status == "BUILD":
+                still_building.append(server)
+            else:
+                raise Exception(
+                    "The resource {} is in an invalid state.".format(
+                        server.id
+                    )
+                )
+        building = still_building
+        counter += delay
+        if counter < timeout:
+            sleep(delay)
+        else:
+            raise Exception(
+                "All resources failed to build within {} seconds.".format(
+                    timeout,
+                )
+            )
+    return active
+
+
+def make_networks(conf):
+    neutron = neutron_client.Client(session=get_session())
+    net = IPv4Network(u'10.0.0.0/8')
+    snets = net.subnets(16)
```

Comment:
I don't think it makes sense to start documenting functionality from other modules.
https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Network.subnets